server/src/test/java/org/spine3/server/storage/RecordStorageShould.java

```diff
@@ -357,6 +362,69 @@ public void write_record_with_columns() {
         assertEquals(record, readRecord.get());
     }
 
+    @SuppressWarnings("OverlyLongMethod") // Complex test case (still tests a single operation)
+    @Test
+    public void filter_records_by_columns() {
+        final Project.Status requiredValue = Project.Status.DONE;
+        final Int32Value wrappedValue = Int32Value.newBuilder()
+                                                  .setValue(requiredValue.getNumber())
+                                                  .build();
+        final Version versionValue = Version.newBuilder()
+                                            .setNumber(2) // Value of the counter after one columns
+                                            .build();     // scan (incremented 2 times internally)
+        final EntityFilters filters = EntityFilters.newBuilder()
+                                                   .putColumnFilter("projectStatusValue",
+                                                                    AnyPacker.pack(wrappedValue))
+                                                   .putColumnFilter("counterVersion",
+                                                                    AnyPacker.pack(versionValue))
+                                                   .build();
+        final EntityQuery query = EntityQueries.from(filters, TestCounterEntity.class);
+        final I idMatching = newId();
+        final I idWrong1 = newId();
+        final I idWrong2 = newId();
+
+        final TestCounterEntity<I> matchingEntity = new TestCounterEntity<>(idMatching);
+        final TestCounterEntity<I> wrongEntity1 = new TestCounterEntity<>(idWrong1);
+        final TestCounterEntity<I> wrongEntity2 = new TestCounterEntity<>(idWrong2);
+
+        // 2 of 3 have required values
+        matchingEntity.setStatus(requiredValue);
+        wrongEntity1.setStatus(requiredValue);
+        wrongEntity2.setStatus(Project.Status.CANCELLED);
+
+        // Change internal Entity state
+        wrongEntity1.getCounter();
+
+        // After the mutation above the single matching record is record under the ID `idMatching`
```

Comment:
`record is record` needs to be fixed.

Also, the ordering should be `under the 'idMatching' ID`.